cmd/admin-subnet-register.go

```diff
@@ -0,0 +1,181 @@
+  {{end}}
+EXAMPLES:
+  1. Register the cluster at 'play' MinIO server.
+     {{.Prompt}} {{.HelpName}} play
+  2. Register the cluster at 'play' MinIO server, naming it as 'play-cluster' on subnet
+     {{.Prompt}} {{.HelpName}} play --name play-cluster
+  3. Register the cluster at 'play' MinIO server, using the proxy 192.168.1.3:3128
+     {{.Prompt}} {{.HelpName}} play --subnet-proxy 192.168.1.3:3128
+`,
+}
+
+// checkAdminRegisterSyntax - validate arguments passed by a user
+func checkAdminRegisterSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) == 0 || len(ctx.Args()) > 1 {
+		cli.ShowCommandHelpAndExit(ctx, "register", 1) // last argument is exit code
+	}
+}
+
+// ClusterRegistrationReq - JSON payload of the subnet api for cluster registration
+// Contains a registration token created by base64 encoding  of the registration info
+type ClusterRegistrationReq struct {
+	Token string `json:"token"`
+}
+
+// ClusterRegistrationInfo - Information stored in the cluster registration token
+type ClusterRegistrationInfo struct {
+	DeploymentID string      `json:"deployment_id"`
+	ClusterName  string      `json:"cluster_name"`
+	UsedCapacity uint64      `json:"used_capacity"`
+	Info         ClusterInfo `json:"info"`
+}
+
+// ClusterInfo - The "info" sub-node of the cluster registration information struct
+// Intended to be extensible i.e. more fields will be added as and when required
+type ClusterInfo struct {
+	MinioVersion    string `json:"minio_version"`
+	NoOfServerPools int    `json:"no_of_server_pools"`
+	NoOfServers     int    `json:"no_of_servers"`
+	NoOfDrives      int    `json:"no_of_drives"`
+	NoOfBuckets     uint64 `json:"no_of_buckets"`
+	NoOfObjects     uint64 `json:"no_of_objects"`
+}
+
+// SubnetLoginReq - JSON payload of the subnet login api
+type SubnetLoginReq struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+// SubnetMFAReq - JSON payload of the subnet mfa api
+type SubnetMFAReq struct {
+	Username string `json:"username"`
+	OTP      string `json:"otp"`
+	Token    string `json:"token"`
+}
+
+func mainAdminRegister(ctx *cli.Context) error {
+	checkAdminRegisterSyntax(ctx)
+
+	offlineMode := ctx.Bool(("offline"))
+	if !offlineMode && !subnetReachable() {
+		console.Fatalln("Subnet not reachable.")
+	}
+
+	// Get the alias parameter from cli
+	aliasedURL := ctx.Args().Get(0)
+
+	// Create a new MinIO Admin Client
+	client := getClient(aliasedURL)
+
+	// Fetch info of all servers (cluster or single server)
+	admInfo, e := client.ServerInfo(globalContext)
+	fatalIf(probe.NewError(e), "Unable to fetch cluster info")
+
+	clusterName := ctx.String("name")
+	if len(clusterName) == 0 {
+		clusterName = aliasedURL
+	}
+
+	regInfo := getClusterRegInfo(admInfo, clusterName)
+
+	if offlineMode {
+		registerOffline(regInfo, aliasedURL)
+	} else {
+		registerOnline(regInfo, aliasedURL, clusterName)
+	}
+
+	msg := fmt.Sprintln("Cluster", aliasedURL, "has been successfully registered.")
```

Comment:
I see the `mc: ` prefix added for these messages. For example

```
mc: Cluster play has been successfully registered.
```
shall we remove it?

Reply:
This looks like a standard for info messages printed by `mc`, and is automatically added by `console.Infoln`